src/engine/Operation.cpp

```diff
@@ -69,6 +69,105 @@ void Operation::recursivelySetTimeConstraint(
   });
 }
 
+// _____________________________________________________________________________
+ProtoResult Operation::runComputation(const ad_utility::Timer& timer,
+                                      ComputationMode computationMode,
+                                      bool isRoot) {
+  checkCancellation();
+  runtimeInfo().status_ = RuntimeInformation::Status::inProgress;
+  signalQueryUpdate();
+  ProtoResult result =
+      computeResult(computationMode == ComputationMode::LAZY_IF_SUPPORTED);
+  AD_CONTRACT_CHECK(computationMode == ComputationMode::LAZY_IF_SUPPORTED ||
+                    result.isDataEvaluated());
+
+  checkCancellation();
+  if constexpr (ad_utility::areExpensiveChecksEnabled) {
+    // Compute the datatypes that occur in each column of the result.
+    // Also assert, that if a column contains UNDEF values, then the
+    // `mightContainUndef` flag for that columns is set.
+    // TODO<joka921> It is cheaper to move this calculation into the
+    // individual results, but that requires changes in each individual
+    // operation, therefore we currently only perform this expensive
+    // change in the DEBUG builds.
+    result.checkDefinedness(getExternallyVisibleVariableColumns());
+  }
+  // Make sure that the results that are written to the cache have the
+  // correct runtimeInfo. The children of the runtime info are already set
+  // correctly because the result was computed, so we can pass `nullopt` as
+  // the last argument.
+  if (result.isDataEvaluated()) {
+    updateRuntimeInformationOnSuccess(result.idTable().size(),
+                                      ad_utility::CacheStatus::computed,
+                                      timer.msecs(), std::nullopt);
+  } else {
+    runtimeInfo().status_ = RuntimeInformation::lazilyMaterialized;
+    result.runOnNewChunkComputed([this, isRoot](
+                                     const IdTable& idTable,
+                                     std::chrono::milliseconds duration) {
+      runtimeInfo().totalTime_ += duration;
+      runtimeInfo().originalOperationTime_ = runtimeInfo().getOperationTime();
+      runtimeInfo().numRows_ += idTable.numRows();
+      runtimeInfo().numCols_ = idTable.numColumns();
+      LOG(DEBUG) << "Computed partial chunk of size " << idTable.numRows()
+                 << " x " << idTable.numColumns() << std::endl;
+      if (isRoot) {
+        signalQueryUpdate();
+      }
```

Comment:
Why not trigger an update for every chunk also in the middle?
I would intuitively drop the `isRoot` here.

Reply:
The idea was to avoid duplicate triggers because if every operation is lazy and all of them don't combine chunks together (which is currently the case for both operations that support it) then most of them would be redundant, but you're probably right.